### PR TITLE
Riccati example refined

### PR DESCRIPTION
**Pull request by andreacamisa**

### CODE REVIEW

File: src/mpc/example.py [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
+
+# =============================
+# begin algorithm
+
 P = np.empty((N, nx, nx))
-K = np.empty((N, nu, nx))
 x = np.empty((N, nx))
 u = np.empty((N, nu))
-print(A)
+x[0, :] = x0
```
Comment (ivanonotarnicola) [inline src/mpc/example.py]: Why not: x[0] = x0?

File: src/mpc/example.py [hunk 2]
```diff
@@ -1,29 +1,62 @@
...
+
+# Section 1.1 at https://notargit.dei.unibo.it/inotarnicola/2019_basic_topics/-/blob/master/LQR%20for%20discrete-time%20systems/notes/main_dLQR.pdf
+
+# backward pass
```
Comment (ivanonotarnicola) [inline src/mpc/example.py:38]: I would add the K as an output of the backward pass.
Then the forward pass uses only the computed gains

File: src/mpc/example.py [hunk 3]
```diff
@@ -1,29 +1,62 @@
...
+    P[k, :, :] = (
+        Q
+        + A.T @ P[k + 1, :, :] @ A
+        - tmp @ np.linalg.inv(R + B.T @ P[k + 1, :, :] @ B) @ tmp.T
+    )
+
+# forward pass
+for k in range(N - 1):
+    K = -np.linalg.inv(R + B.T @ P[k + 1, :, :] @ B) @ (B.T @ P[k + 1, :, :] @ A)
+    u[k, :] = K @ x[k, :]
```
Comment (ivanonotarnicola) [inline src/mpc/example.py]: u[k] = K @ x[k]

File: src/mpc/example.py [hunk 4]
```diff
@@ -1,29 +1,62 @@
...
-for k in reversed(range(N)):
-    P[k] = A.T @ P[k + 1] @ A
-    K[k] = A @ x[k] + B @ u[k]
+print("Solution:")
 
 for k in range(N):
-    u[k] = K[k] @ x[k]
-    x[k + 1] = A @ x[k] + B @ u[k]
+    print(f"time {k}")
+    print(f"\tx = {np.array2string(x[k, :], precision=4, floatmode='fixed')}")
```
Comment (ivanonotarnicola) [inline src/mpc/example.py]: BANG! array2string è strong

File: src/mpc/example.py (hunk 3 above)
Comment (ivanonotarnicola) [reply src/mpc/example.py]: Daje

Comment (andreacamisa) [reply src/mpc/example.py]: Done

File: src/mpc/example.py (hunk 1 above)
Comment (andreacamisa) [reply src/mpc/example.py]: Done

File: src/mpc/example.py (hunk 2 above)
Comment (andreacamisa) [reply src/mpc/example.py:38]: Let's avoid storing the K for now

File: src/mpc/example.py (hunk 4 above)
Comment (andreacamisa) [reply src/mpc/example.py]: I know

**Review by ivanonotarnicola — APPROVED** [review]

**Pull request merged**